viadot/flows/mediatool_to_adls.py

```diff
@@ -0,0 +1,171 @@
+
+        self.local_file_path = (
+            local_file_path or self.slugify(name) + self.output_file_extension
+        )
+        self.local_json_path = self.slugify(name) + ".json"
+        self.adls_dir_path = adls_dir_path
+
+        if adls_file_name is not None:
+            self.adls_file_path = os.path.join(adls_dir_path, adls_file_name)
+            self.adls_schema_file_dir_file = os.path.join(
+                adls_dir_path, "schema", Path(adls_file_name).stem + ".json"
+            )
+        else:
+            self.adls_file_path = os.path.join(
+                adls_dir_path, self.now + self.output_file_extension
+            )
+            self.adls_schema_file_dir_file = os.path.join(
+                adls_dir_path, "schema", self.now + ".json"
+            )
+
+        super().__init__(*args, name=name, **kwargs)
+
+        self.gen_flow()
+
+    @staticmethod
+    def slugify(name):
+        return name.replace(" ", "_").lower()
+
+    def gen_flow(self) -> Flow:
+        mediatool_to_df_task = MediatoolToDF(
+            mediatool_credentials=self.mediatool_credentials,
+            mediatool_credentials_key=self.mediatool_credentials_key,
+        )
+
+        df = mediatool_to_df_task.bind(
+            organization_ids=self.organization_ids,
+            media_entries_columns=self.media_entries_columns,
+            flow=self,
+        )
+
+        df_with_metadata = add_ingestion_metadata_task.bind(df, flow=self)
+        dtypes_dict = df_get_data_types_task.bind(df_with_metadata, flow=self)
+
+        if self.output_file_extension == ".parquet":
+            df_to_be_loaded = df_map_mixed_dtypes_for_parquet(
+                df_with_metadata, dtypes_dict, flow=self
+            )
+            df_to_file = df_to_parquet.bind(
+                df=df_to_be_loaded,
+                path=self.local_file_path,
+                if_exists=self.if_exists,
+                flow=self,
+            )
+        else:
+            df_to_file = df_to_csv.bind(
+                df=df_with_metadata,
+                path=self.local_file_path,
+                if_exists=self.if_exists,
+                flow=self,
+            )
+
+        file_to_adls_task.bind(
+            from_path=self.local_file_path,
+            to_path=self.adls_file_path,
+            overwrite=self.overwrite,
+            sp_credentials_secret=self.adls_sp_credentials_secret,
+            flow=self,
+        )
+
+        dtypes_updated = update_dtypes_dict(dtypes_dict, flow=self)
+        dtypes_to_json_task.bind(
+            dtypes_dict=dtypes_updated, local_json_path=self.local_json_path, flow=self
+        )
+
+        json_to_adls_task.bind(
+            from_path=self.local_json_path,
+            to_path=self.adls_schema_file_dir_file,
+            overwrite=self.overwrite,
+            sp_credentials_secret=self.adls_sp_credentials_secret,
+            flow=self,
+        )
+
+        df_with_metadata.set_upstream(df, flow=self)
```

Comment:
are all those upstreams really needed? for example df_with_metadata uses df so upstream is not needed

Reply:
I added them to have the correct task queue